### PR TITLE
fix(lily): Consider partially completed heights w ERRORs and SKIPs

### PR DESCRIPTION
**Pull request by placer14**
fixes https://github.com/filecoin-project/lily/issues/773
fixes https://github.com/filecoin-project/lily/issues/775

### CODE REVIEW

File: chain/find.go [hunk 1]
```diff
@@ -213,31 +213,41 @@ func (g *GapIndexer) findTaskEpochGaps(ctx context.Context) (visor.GapReportList
...
+			where status = ?5
+			group by task_height_count.height, task_height_count.status
+		) task_count_per_height
+		where task_count_per_height.cheight != ?0
+		and status = ?5
+	) incomplete
+	on vpr.height = incomplete.height
+	where (vpr.status_information != ?1
+	or vpr.status_information is null)
+	and vpr.status = ?5
```
Comment (frrist) [inline chain/find.go:239]: fixes https://github.com/filecoin-project/lily/issues/773?

File: chain/find.go [hunk 2]
```diff
@@ -213,31 +213,41 @@ func (g *GapIndexer) findTaskEpochGaps(ctx context.Context) (visor.GapReportList
...
+) incomplete_heights_and_their_completed_tasks
 order by height desc
+;
 `,
-		len(AllTasks), visor.ProcessingStatusInformationNullRound, g.minHeight, g.maxHeight,
+		len(AllTasks), // arg 0
+		visor.ProcessingStatusInformationNullRound, // arg 1
+		g.minHeight, // arg 2
+		g.maxHeight, // arg 3
+		visor.ProcessingStatusError, // arg 4
```
Comment (frrist) [inline chain/find.go:249]: Marks errors as gaps?

File: chain/find.go (hunk 1 above)
Comment (placer14) [reply chain/find.go:240]: Correct.

Comment (placer14) [reply chain/find.go]: I've added notes within the query to explain what's happening here.

File: chain/find.go (hunk 2 above)
Comment (placer14) [reply chain/find.go]: I've added notes within the query to explain what's happening [here](https://github.com/filecoin-project/lily/pull/791/files#diff-d49953a87209cc88edee91a3c0173f069297d6c4835cd1e17340094e4fbfd94aR217-R251).

**Review by frrist — APPROVED** [review]
Good stuff, thanks for fixing these

**Pull request merged**